AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/SubscriptionSync/IncomingAsyncSubscriptionEventPublisher.swift

```diff
@@ -81,56 +81,47 @@ final class IncomingAsyncSubscriptionEventPublisher: Cancellable {
 
     func onCreateListenerHandler(event: Event) {
         log.verbose("onCreateListener: \(event)")
-        if case .inProcess(.connection) = event {
-            let onCreateConnectionOp = CancelAwareBlockOperation {
-                if case .inProcess(.connection(.connected)) = event {
-                    self.onCreateConnected = true
-                } else if case .inProcess(.connection(.disconnected)) = event {
-                    self.onCreateConnected = false
-                }
-                if self.combinedConnectionStatus {
-                    self.incomingSubscriptionEvents.send(event)
-                }
-            }
-            self.connectionStatusQueue.addOperation(onCreateConnectionOp)
-        } else {
-            incomingSubscriptionEvents.send(event)
+        let onCreateConnectionOp = CancelAwareBlockOperation {
+            self.onCreateConnected = self.determineConnectionStatusOn(event: event)
+            self.sendConnectionEventIfConnected(event: event)
         }
+        genericListenerHandler(event: event, cancelAwareBlock: onCreateConnectionOp)
     }
-    
+
     func onUpdateListenerHandler(event: Event) {
         log.verbose("onUpdateListener: \(event)")
-        if case .inProcess(.connection) = event {
-            let onUpdateConnectionOp = CancelAwareBlockOperation {
-                if case .inProcess(.connection(.connected)) = event {
-                    self.onUpdateConnected = true
-                } else if case .inProcess(.connection(.disconnected)) = event {
-                    self.onUpdateConnected = false
-                }
-                if self.combinedConnectionStatus {
-                    self.incomingSubscriptionEvents.send(event)
-                }
-            }
-            self.connectionStatusQueue.addOperation(onUpdateConnectionOp)
-        } else {
-            incomingSubscriptionEvents.send(event)
+        let onUpdateConnectionOp = CancelAwareBlockOperation {
+            self.onUpdateConnected = self.determineConnectionStatusOn(event: event)
+            self.sendConnectionEventIfConnected(event: event)
         }
+        genericListenerHandler(event: event, cancelAwareBlock: onUpdateConnectionOp)
     }
 
     func onDeleteListenerHandler(event: Event) {
         log.verbose("onDeleteListener: \(event)")
+        let onDeleteConnectionOp = CancelAwareBlockOperation {
+            self.onDeleteConnected = self.determineConnectionStatusOn(event: event)
+            self.sendConnectionEventIfConnected(event: event)
+        }
+        genericListenerHandler(event: event, cancelAwareBlock: onDeleteConnectionOp)
+    }
+
+    func determineConnectionStatusOn(event: Event) -> Bool {
+        if case .inProcess(.connection(.connected)) = event {
+            return true
+        }
+        return false
+    }
+
+    func sendConnectionEventIfConnected(event: Event) {
+        if combinedConnectionStatus {
```

Comment:
I don't immediately see where this is named, but again, Booleans should be named with an `isXXX` pattern.

Reply:
This is a computed property located on line 38, and in this case, i don't agree with following the isXXX pattern.  That being said, I think the name could be better, so I changed this to:
````
combinedConnectionStatusIsConnected
```

Reply:
To be clear, i think the `isXXX` pattern works well for singular cases (and thanks for calling it out above), but in this case, we are referring to a case which is computed from multiple components, and the wording of:
```
if isCombinedConnectionStatus {
}
```
seems less clear than
```
if combinedConnectionStatusIsConnected
```

At this point, I'm going to bias for action and push in since I have approval, with the understanding that I'm open to debate on changing the name at any given time.